pymongo/uri_parser.py

```diff
@@ -67,6 +67,8 @@ def parse_userinfo(userinfo):
     # No password is expected with GSSAPI authentication.
     if not user:
         raise InvalidURI("The empty string is not valid username.")
+    if '+' in user or '+' in passwd:
+        warnings.warn("Starting in 4.0 PyMongo will no longer decode plus signs in user names or passwords to spaces", stacklevel=2)
```

Comment:
"user names" -> "usernames"

Can we including something like this in the warning: "use quote() to escape the username/password instead of quote_plus()"

Reply:
Also I'm curious about the stacklevel. What does the Warning look like for MongoClient(uri) and for parse_uri(uri)?